### PR TITLE
keyboard spec

### PR DESCRIPTION
**Pull request by pelikhan**
HID keyboard spec

### CODE REVIEW

File: services/keyboard.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    identifier: 0x18b05b6a
+    
+A keyboard that can send keystrokes.
+
+The codes for the key (selectors) is defined in the [HID Keyboard
+specification](https://usb.org/sites/default/files/hut1_21.pdf), chapter 10 Keyboard/Keypad Page, page 81.
+    
+## Commands
+
+    flags Modifiers : u8 {
```
Comment (mmoskal) [inline services/keyboard.md:12]: you would need to use different numbers (separate bits) to make this flags; given that there's already 8 of them, I suggest u16

File: services/keyboard.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        RightGUID = 0xe3
+    }
+    enum Action : u8 {
+        Press = 0
+        Up = 1
+        Down = 2
+    }
+    command key @ 0x80 {
+    repeats:
+        selector: u8
```
Comment (mmoskal) [inline services/keyboard.md:29]: selector is 16 bit in the HID spec (though they only define codes until 0xE8)

File: services/keyboard.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Keyboard
+
+    identifier: 0x18b05b6a
+    
+A keyboard that can send keystrokes.
+
+The codes for the key (selectors) is defined in the [HID Keyboard
+specification](https://usb.org/sites/default/files/hut1_21.pdf), chapter 10 Keyboard/Keypad Page, page 81.
```
Comment (mmoskal) [inline services/keyboard.md:8]: I think we should repeat the list here...

File: services/keyboard.md (hunk 1 above)
Comment (mmoskal) [reply services/keyboard.md:12]: do they even treat modifiers separately in the HID spec? maybe we also don't have to

Comment (mmoskal) [reply services/keyboard.md:12]: oh and the key is GUI not GUID

File: services/keyboard.md (hunk 3 above)
Comment (pelikhan) [reply services/keyboard.md:8]: it's like 5 pages long and anyone implementing HID will have to lookup that table.

File: services/keyboard.md (hunk 1 above)
Comment (pelikhan) [reply services/keyboard.md:12]: Ooops, copy paste issue. The HID spec reserves a single range for the modifiers so u8 is enough

**Pull request merged**